[CX2] Graphviz layout errors out on GO-CAM with quote in title

The title of `gomodel:666b894f00001546` is `HTR4-mediated serotonin signaling G-alpha-s ("stimulatory", activating adenylate cyclase) (Human).` When converting to CX2 and applying a Graphviz layout the underlying `dot` command errors out. The error message seems to indicate that the quote is causing the issues:

```bash
$ gocam fetch 666b894f00001546 | gocam convert -O cx2 -I yaml --dot-layout
"dot" with args ['-Tdot', '/var/folders/92/1q4rzv253mvcys9r88vg_mgc0000gp/T/tmpdap6x0zq/tmpcsddjp0s'] returned code: 1

stdout, stderr:
 b''
b"Error: /var/folders/92/1q4rzv253mvcys9r88vg_mgc0000gp/T/tmpdap6x0zq/tmpcsddjp0s: syntax error in line 1 near 'stimulatory'\n"

AssertionError: "dot" with args ['-Tdot', '/var/folders/92/1q4rzv253mvcys9r88vg_mgc0000gp/T/tmpdap6x0zq/tmpcsddjp0s'] returned code: 1
```

--- a/src/gocam/translation/cx2/main.py
+++ b/src/gocam/translation/cx2/main.py
@@ -301,8 +301,9 @@ def model_to_cx2(
         # Convert the CX2 network to a networkx graph
         networkx_graph = CX2NetworkXFactory().get_graph(cx2_network)
 
-        # Our node and edge attributes confuse the pydot conversion, but we don't need them just
-        # for layout
+        # Our graph, node, and edge attributes confuse the pydot conversion, but we don't need them
+        # just for layout
+        networkx_graph.graph.clear()
         for node_id in networkx_graph.nodes:
             networkx_graph.nodes[node_id].clear()
         for edge_id in networkx_graph.edges:

--- a/src/gocam/translation/cx2/main.py
+++ b/src/gocam/translation/cx2/main.py
@@ -96,17 +96,31 @@ def model_to_cx2(
             return object_id
         return _remove_species_code_suffix(object.label)
 
+    def _format_curie_link(curie: str) -> str:
+        try:
+            url = go_converter.expand(curie)
+            return _format_link(url, curie)
+        except ValueError:
+            return curie
+
     def _format_evidence_list(evidence_list: List[EvidenceItem]) -> str:
         """Format a list of evidence items as an HTML unordered list."""
         evidence_list_items = []
         for e in evidence_list:
-            reference_link = _format_link(go_converter.expand(e.reference), e.reference)
-            evidence_item = f"{reference_link} ({_get_object_label(e.term)})"
+            evidence_item = ""
+            if e.reference:
+                evidence_item += _format_curie_link(e.reference)
+            if e.term:
+                term_label = _get_object_label(e.term)
+                if evidence_item:
+                    evidence_item += f" ({term_label})"
+                else:
+                    evidence_item += term_label
             if e.with_objects:
-                with_objects = ", ".join(
-                    _format_link(go_converter.expand(o), o) for o in e.with_objects
-                )
-                evidence_item += f" with/from {with_objects}"
+                with_objects = ", ".join(_format_curie_link(o) for o in e.with_objects)
+                if evidence_item:
+                    evidence_item += " "
+                evidence_item += f"with/from {with_objects}"
             evidence_list_items.append(f"<li>{evidence_item}</li>")
         return f'<ul style="padding-inline-start: 1rem">{"".join(evidence_list_items)}</ul>'
 

